scripts/build.sh

```diff
@@ -106,11 +107,16 @@ main() {
       portable=""
   fi
 
+  if [[ $platform == "macos" && $arch == "arm64" ]]; then
+      cross_file="--cross-file resources/macos/macos_arm64.conf"
+  fi
```

Comment:
Shouldn't this be applied only when cross compiling?

Reply:
Changing `$arch` with `$CROSS_ARCH` would do. But also we should tell people to use meson directly instead of the shell scripts to build, since the shell scripts are mostly a CI thing and building with meson should be straight forward.

Reply:
Could we get rid of the shell scripts completely and replace them with meson?

Reply:
the shell scripts make the yaml files smaller and cleaner since a lot of the build+packaging combo logic is performed on them which we would need to replicate on the CI yaml files. For the moment I will just replace $arch with $CROSS_ARCH, later we can tinker with build+packaging simplifying ideas